staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -6145,9 +6145,10 @@ type NodeAddressType string
 
 // These are built-in addresses type of node. A cloud provider may set a type not listed here.
 const (
-	// NodeHostName identifies a name of the node. Although every node can be assumed
-	// to have a NodeAddress of this type, its exact syntax and semantics are not
-	// defined, and are not consistent between different clusters.
+	// NodeHostName identifies the node name of the system. Although every node can be
+	// assumed to have a NodeAddress of this type, its exact syntax and semantics are
+	// not defined, and are not consistent between different clusters. It is recommended
+	// to use the system node name reported by kernel for this field.
```

Comment:
The point of saying that the semantics are not defined is that you can't rely on this value being useful for anything. Given that, it doesn't seem like it makes sense to recommend that it be set in any particular way, because there are going to be legacy clouds that don't set it that way, so you still can't use it for anything.

If anything, we should be clearer about the fact that nobody should ever look at this field. Maybe we should mark it deprecated.

Reply:
you can not deprecate this field, is used by the certificate manager and people can be using it.

Reply:
If our recommendation here is going to be constrained by the fact that kubelet assumes that the value is a DNS name pointing to the node, then we can't recommend that users set it to "`the system node name reported by kernel`", since that is not guaranteed to be a DNS name pointing to the node. Right?

This is what I mean about "you can't rely on this value being useful for anything". If you recommend that people use it in a way that unbreaks #126088, then you break the certificate requests for some users. The right fix is that neither the certificate request nor the e2e tests should be relying on this value, because sometimes it's one thing, and sometimes it's another.

(In fact, kubelet doesn't "rely" on this value; it puts the `NodeHostName` value into the certificate request as either an IP address or a DNS name (whichever one it parses as), but it doesn't _need_ it; it will also take IPs from the `InternalIP` and `ExternalIP` values, and DNS names from the `InternalDNS` and `ExternalDNS` values.)

Reply:
Is about `hostname` node.status.address matching the reported hostname by the kernel so node authorizers can confirm the certificate request is valid, ... DNSname here is a certificate terminology, not about the DNS record
Not a big deal, just thought could be interesting reflecting this , I'm fine with closing